projects/ng2-smart-table/src/lib/lib/grid.ts

```diff
@@ -268,4 +268,33 @@ export class Grid {
     return this.dataSet.getLastRow();
   }
 
+  private getSelectionInfo(): { perPage: number, page: number, selectedRowIndex: number, switchPageToSelectedRowPage: boolean } {
+    const switchPageToSelectedRowPage: boolean = this.getSetting('switchPageToSelectedRowPage');
+    const selectedRowIndex: number = Number(this.getSetting('selectedRowIndex', 0)) || 0;
```

Comment:
Number function is for convert possible string value in settings, and || 0 for default 0 if Number will return NaN.
and because NaN we use || 0 instead of the default value

Reply:
Ok, but what is the reason for the `Number` function to return `NaN`? Incorrect value of `selectedRowIndex` settings? Shouldn't we throw an error informing the developer about misconfiguration?